src/Umbraco.Core/Manifest/IManifestFilter.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections.Generic;
+
+namespace Umbraco.Core.Manifest
+{
+    /// <summary>
+    /// Provides filtering for package manifests.
+    /// </summary>
+    public interface IManifestFilter
+    {
+        /// <summary>
+        /// Filters package manifests.
+        /// </summary>
+        /// <param name="manifests">The package manifests.</param>
+        /// <remarks>
+        /// <para>It is possible to remove, change, or add manifests.</para>
+        /// </remarks>
+        void Filter(List<PackageManifest> manifests);
```

Comment:
Why is `List<>` being used here?

Reply:
Because we want the filters to be able to add, remove, change... etc the list of manifests. A `List` was simple. Each filter can alter the list. But we can switch to an `IEnumerable` if you'd rather have it an `IEnumerable` - makes each filter a bit more convoluted but would work.

Reply:
Would it make more sense to have the method return the result of the filter ? so like `IEnumerable<PackageManifest> Filter(IEnumerable<PackageManifest> manifests)` ?

Reply:
Can do, does not make a lot of difference I believe. Want me to?

Reply:
no it's fine, just had to go look at it's usage.